packages/luals-annotations/src/annotationBuilder.ts

```diff
@@ -0,0 +1,127 @@
+import luaLSType, { formatProperty } from "./luaLSType";
+import { Function, Property } from "@lest/docs";
+
+type FunctionRenderOptions = {
+	staticMethod?: boolean;
+	className?: string;
+};
+
+function renderParameterAnnotations(parameters: Property[]): string[] {
+	return parameters.map((param) => `---@param ${formatProperty(param, " ")}`);
+}
+
+function renderReturnAnnotations(returns: Property[]): string[] {
+	return returns.map((ret) => `---@return ${luaLSType(ret)} ${ret.name ?? ""}`);
+}
+
+function renderFunctionAliases(func: Function, staticMethod: boolean, className: string): string[] {
+	const { parameters = [] } = func;
+
+	const functionCharacter = staticMethod ? "." : ":";
+	const functionPrefix = className ? `${className}${functionCharacter}` : "";
+
+	const functionNames = [func.name, ...(func.aliases ?? [])];
+
+	return functionNames.map(
+		(name) => `function ${functionPrefix}${name}(${parameters.map((param) => param.name)}) end`
+	);
+}
```

Comment:
This code is quite difficult to read. A lot of the variables don't make sense, or don't need to exist, and some things that aren't variables (like mapping the parameters) should be.